### PR TITLE
Rename bindings, docs, shared libraries, headers

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

**Comment by reuben** [comment]
@lissyx It's pretty much impossible to review the diff itself I guess, but would be great if you could double check the built artifacts to see if you catch something.

File: doc/BUILDING.rst [hunk 1]
```diff
@@ -123,7 +123,7 @@ Included are a set of generated Python bindings. After following the above build
    make bindings
    pip install dist/deepspeech*
```
Comment (lissyx) [inline doc/BUILDING.rst:124]: forgot this one?

File: doc/BUILDING.rst [hunk 2]
```diff
@@ -278,7 +278,7 @@ Running ``deepspeech`` via adb
 You should use ``adb push`` to send data to device, please refer to Android
 documentation on how to use that.
 
-Please push DeepSpeech data to ``/sdcard/deepspeech/``\ , including:
+Please push Mozilla Voice STT data to ``/sdcard/deepspeech/``\ , including:
```
Comment (lissyx) [inline doc/BUILDING.rst:281]: we should change the path here as well?

File: doc/BUILDING.rst [hunk 3]
```diff
@@ -289,7 +289,7 @@ Please push DeepSpeech data to ``/sdcard/deepspeech/``\ , including:
 Then, push binaries from ``native_client.tar.xz`` to ``/data/local/tmp/ds``\ :
```
Comment (lissyx) [inline doc/BUILDING.rst:289]: and ds -> mvstt

File: doc/TRAINING.rst [hunk 4]
```diff
@@ -32,14 +32,14 @@ Once this command completes successfully, the environment will be ready to be ac
 Activating the environment
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Each time you need to work with DeepSpeech, you have to *activate* this virtual environment. This is done with this simple command:
+Each time you need to work with Mozilla Voice STT, you have to *activate* this virtual environment. This is done with this simple command:
 
 .. code-block::
 
    $ source $HOME/tmp/deepspeech-train-venv/bin/activate
```
Comment (lissyx) [inline doc/TRAINING.rst:39]: should we change this ?

File: doc/USING.rst [hunk 5]
```diff
@@ -93,16 +93,16 @@ Once this command completes successfully, the environment will be ready to be ac
 Activating the environment
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Each time you need to work with DeepSpeech, you have to *activate* this virtual environment. This is done with this simple command:
+Each time you need to work with Mozilla Voice STT, you have to *activate* this virtual environment. This is done with this simple command:
 
 .. code-block::
 
    $ source $HOME/tmp/deepspeech-venv/bin/activate
```
Comment (lissyx) [inline doc/USING.rst:100]: same, should we change this?

File: doc/USING.rst [hunk 6]
```diff
@@ -192,7 +192,7 @@ also, if you need some binaries different than current master, like ``v0.2.0-alp
 
    python3 util/taskcluster.py --branch "v0.2.0-alpha.6" --target "."
 
-The script ``taskcluster.py`` will download ``native_client.tar.xz`` (which includes the ``deepspeech`` binary and associated libraries) and extract it into the current folder. Also, ``taskcluster.py`` will download binaries for Linux/x86_64 by default, but you can override that behavior with the ``--arch`` parameter. See the help info with ``python util/taskcluster.py -h`` for more details. Specific branches of DeepSpeech or TensorFlow can be specified as well.
+The script ``taskcluster.py`` will download ``native_client.tar.xz`` (which includes the ``deepspeech`` binary and associated libraries) and extract it into the current folder. Also, ``taskcluster.py`` will download binaries for Linux/x86_64 by default, but you can override that behavior with the ``--arch`` parameter. See the help info with ``python util/taskcluster.py -h`` for more details. Specific branches of Mozilla Voice STT or TensorFlow can be specified as well.
```
Comment (lissyx) [inline doc/USING.rst:195]: ``deepspeech`` -> ``mozilla_voice_stt`` ?

File: doc/conf.py [hunk 7]
```diff
@@ -41,7 +41,7 @@
 
 # -- Project information -----------------------------------------------------
 
-project = u'DeepSpeech'
+project = u'Mozilla Voice STT'
 copyright = '2019-2020, Mozilla Corporation'
```
Comment (lissyx) [inline doc/conf.py:45]: nit: this copyright date range is wrong :)

File: doc/index.rst [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
-DeepSpeech is an open source Speech-To-Text engine, using a model trained by machine learning techniques based on `Baidu's Deep Speech research paper <https://arxiv.org/abs/1412.5567>`_. Project DeepSpeech uses Google's `TensorFlow <https://www.tensorflow.org/>`_ to make the implementation easier.
+Mozilla Voice STT is an open source Speech-To-Text engine, using a model trained by machine learning techniques based on `Baidu's Deep Speech research paper <https://arxiv.org/abs/1412.5567>`_. Project Mozilla Voice STT uses Google's `TensorFlow <https://www.tensorflow.org/>`_ to make the implementation easier.
 
-To install and use DeepSpeech all you have to do is:
+To install and use Mozilla Voice STT all you have to do is:
 
 .. code-block:: bash
 
    # Create and activate a virtualenv
    virtualenv -p python3 $HOME/tmp/deepspeech-venv/
```
Comment (lissyx) [inline doc/index.rst:16]: we should rename the virtualenv

File: doc/index.rst [hunk 9]
```diff
@@ -40,7 +40,7 @@ Quicker inference can be performed using a supported NVIDIA GPU on Linux. See th
    virtualenv -p python3 $HOME/tmp/deepspeech-gpu-venv/
```
Comment (lissyx) [inline doc/index.rst:40]: virtualenv rename?

File: evaluate_tflite.py [hunk 10]
```diff
@@ -19,8 +19,8 @@
 
 r'''
```
Comment (lissyx) [inline evaluate_tflite.py:20]: this doc is old and wrong, we should remove it and instruct to `pip install deepspeech` or `pip install deepspeech-tflite` (or the new package name)

File: native_client/Makefile [hunk 11]
```diff
@@ -16,32 +16,32 @@ include definitions.mk
 default: $(DEEPSPEECH_BIN)
```
Comment (lissyx) [inline native_client/Makefile:16]: maybe we should rename `DEEPSPEECH_BIN` now?

File: native_client/java/app/build.gradle [hunk 12]
```diff
@@ -4,7 +4,7 @@ android {
     compileSdkVersion 27
 
     defaultConfig {
-        applicationId "org.mozilla.deepspeech"
+        applicationId "org.mozilla.voice.sttapp"
```
Comment (lissyx) [inline native_client/java/app/build.gradle:7]: nit: shouldnt this be `org.mozilla.voice.stt.app` if we follow the package?

File: native_client/java/libmozillavoicestt/src/main/java/org/mozilla/voice/stt/MozillaVoiceSttStreamingState.java [hunk 13]
```diff
@@ -0,0 +1,13 @@
+package org.mozilla.voice.stt;
```
Comment (lissyx) [inline native_client/java/libmozillavoicestt/src/main/java/org/mozilla/voice/stt/MozillaVoiceSttStreamingState.java:1]: Why is this file added? It looks like to be SWIG-generated

File: native_client/javascript/Makefile [hunk 14]
```diff
@@ -2,7 +2,7 @@ NODE_BUILD_TOOL    ?= node-pre-gyp
 NODE_ABI_TARGET    ?= 
 NODE_BUILD_VERBOSE ?= --verbose
 NPM_TOOL           ?= npm
-PROJECT_NAME       ?= deepspeech
+PROJECT_NAME       ?= mozilla_voice_stt
```
Comment (lissyx) [inline native_client/javascript/Makefile:5]: this makes me thinking, we used to `-gpu` or `-tflite`, are we going to `_gpu` or `_tflite` ?

Also, side note, but sicne we are breaking compat, can we name `cuda` instead of `gpu` ? We can still hope to have OpenCL or ROCm support, this last one seems to have got some traction on TensorFlow side

File: native_client/python/setup.py [hunk 15]
```diff
@@ -24,7 +24,7 @@ def read(fname):
     numpy_include = os.getenv('NUMPY_INCLUDE', numpy_include)
     numpy_min_ver = os.getenv('NUMPY_DEP_VERSION', '')
 
-    project_name = 'deepspeech'
+    project_name = 'mozilla_voice_stt'
```
Comment (lissyx) [inline native_client/python/setup.py:27]: same question as for node package

File: native_client/Makefile (hunk 11 above)
Comment (reuben) [reply native_client/Makefile:16]: I'm leaving implementation stuff to be renamed later to avoid making this even more hairy.

File: native_client/java/app/build.gradle (hunk 12 above)
Comment (reuben) [reply native_client/java/app/build.gradle:7]: I don't think it makes sense for the example app to be a subpackage of the binding.

File: taskcluster/tc-build-utils.sh [hunk 16]
```diff
@@ -17,9 +17,9 @@ do_deepspeech_python_build()
 
   SETUP_FLAGS=""
   if [ "${package_option}" = "--cuda" ]; then
-    SETUP_FLAGS="--project_name deepspeech-gpu"
+    SETUP_FLAGS="--project_name mozilla_voice_stt-gpu"
```
Comment (lissyx) [inline taskcluster/tc-build-utils.sh:20]: so we mix? not a big fan :)
also, maybe we should name `-gpu` as `-cuda`

File: native_client/javascript/Makefile (hunk 14 above)
Comment (reuben) [reply native_client/javascript/Makefile:5]: Currently we're sticking with `-gpu` and `-tflite`. I don't have strong opinions on going either way.

File: taskcluster/tc-py-utils.sh [hunk 17]
```diff
@@ -286,17 +286,17 @@ get_python_pkg_url()
 get_tflite_python_pkg_name()
 {
   # Default to deepspeech package
-  local _pkgname="deepspeech_tflite"
+  local _pkgname="mozilla_voice_stt_tflite"
```
Comment (lissyx) [inline taskcluster/tc-py-utils.sh:289]: so here we have underscore where elsewhere you set dash :)

File: taskcluster/tc-build-utils.sh (hunk 16 above)
Comment (reuben) [reply taskcluster/tc-build-utils.sh:20]: I actually like it a bit to separate the project name from the package type modifier. But I can change to `_gpu`/`_cuda` if you prefer.

File: native_client/java/libmozillavoicestt/src/main/java/org/mozilla/voice/stt/MozillaVoiceSttStreamingState.java (hunk 13 above)
Comment (reuben) [reply native_client/java/libmozillavoicestt/src/main/java/org/mozilla/voice/stt/MozillaVoiceSttStreamingState.java:1]: https://github.com/mozilla/DeepSpeech/blob/master/native_client/java/libdeepspeech/src/main/java/org/mozilla/deepspeech/libdeepspeech/DeepSpeechStreamingState.java

File: taskcluster/tc-build-utils.sh (hunk 16 above)
Comment (lissyx) [reply taskcluster/tc-build-utils.sh:20]: well, the thing is we won't be able to ship cuda + rocm or + opencl under the same package, so it's better we take the opportunity we have to rename to ease our future

File: native_client/Makefile (hunk 11 above)
Comment (lissyx) [reply native_client/Makefile:16]: ok, it's just that you already went through the trouble of adding missing `DEEPSPEECH_BIN` :)

File: taskcluster/tc-py-utils.sh (hunk 17 above)
Comment (reuben) [reply taskcluster/tc-py-utils.sh:289]: Huh. Looks like Python/pip normalizes dashes to underscores anyway.

File: native_client/java/libmozillavoicestt/src/main/java/org/mozilla/voice/stt/MozillaVoiceSttStreamingState.java (hunk 13 above)
Comment (lissyx) [reply native_client/java/libmozillavoicestt/src/main/java/org/mozilla/voice/stt/MozillaVoiceSttStreamingState.java:1]: maybe it was my mistake :)

File: taskcluster/tc-build-utils.sh (hunk 16 above)
Comment (reuben) [reply taskcluster/tc-build-utils.sh:20]: Going with underline, and `_cuda` then.

**Comment by reuben** [comment]
And we have liftoff: ae9fdb183ec6eb422635c0e3a44c0c2ee5732224
